sdl-core/src/test/scala/io/smartdatalake/app/SmartDataLakeBuilderTest.scala

```diff
@@ -491,13 +491,22 @@ class SmartDataLakeBuilderTest extends FunSuite with BeforeAndAfter {
     val dfResult2 = tgt1DO.getSparkDataFrame(Seq())
     assert(dfResult2.select(functions.max($"nb".cast("int")), functions.count("*")).as[(Int,Long)].head == (20,20))
 
-    // start 3rd dag run
+    // start 3rd dag run -> no data
     action1.reset
     sdlb.run(sdlConfig)
 
     // check results
     val dfResult3 = tgt1DO.getSparkDataFrame(Seq())
-    assert(dfResult3.select(functions.max($"nb".cast("int")), functions.count("*")).as[(Int,Long)].head == (30,30))
+    assert(dfResult3.select(functions.max($"nb".cast("int")), functions.count("*")).as[(Int,Long)].head == (20,20))
+
+    // start 4th dag run
+    // TODO: keep state when skipped!
```

Comment:
you have already implemented this TODO in io.smartdatalake.workflow.ActionDAGRun#getRuntimeInfos

Reply:
Good catch!